src/common/KokkosKernels_PrintUtils.hpp

```diff
@@ -132,6 +132,26 @@ inline void kk_print_1Dview(std::ostream& os, idx_array_type view,
   }
 }
 
+/**
+ * \brief Multi-vector variant (see rank-1 for param description). Prints Nx1
+ * rank-2 vectors same like rank-1 vectors and prints multi-vector dimensions.
+ */
+template <typename idx_array_type>
+inline std::enable_if_t<idx_array_type::rank != 1> kk_print_1Dview(
```

Comment:
Maybe change to `idx_array_type::rank > 1`, I think there is such a thing as a rank 0 view...

Reply:
That's a good idea. The rank-1 version above could be compatible with rank-0 views if you replaced ``host_view(i)`` with ``host_view.access(i)`` (where i still loops from 0 to extent(0)).

Reply:
Thank you!
How about promoting rank-0 view into rank-1 one explicitly ? (please see the fix applied)

Reply:
OK, this way of handling rank-0 is fine. I missed this on the first review, but if you're going to do create ``rank1_view(view.data(), n)`` from a rank-2 view, you should also check that ``view.stride(0) == 1``. Even if it has only 1 column the elements to print are not necessarily contiguous.

Reply:
Actually, it might be easier to just print in a way that works for ranks 0-2 regardless of layout/strides:
```
if(idx_array_type::rank == 2 && view.extent(1) == 1) ||
      idx_array_type::rank == 0)

for(i = 0; i < v.extent(0); i++)
  for(j = 0; j < v.extent(1); j++)
    os << v.access(i, j) << ' ';
```
since access() lets you pass in more indices than the rank, as long as they are in-bounds (0).

Reply:
Right, thank you @brian-kelley: I'm bringing back `subview(view, Kokkos::ALL, 0)` for rank-2 (to address possible discontinuity) and switching to `access()` for correct handling rank-0. Main routine becomes `rank<=1` variant and "the dispatcher" becomes `rank>=2` variant as @lucbv suggested in the first place.